lib/sudo-pam/src/rpassword.rs

```diff
@@ -0,0 +1,105 @@
+/// Derived work from rpassword and rtoolbox, Copyright (c) 2023, Conrad Kleinespel et al
+///
+/// This module contains code that was originally written by Conrad Kleinespel for the rpassword
+/// crate. No copyright notices were found in the original code.
+
+/// See: https://docs.rs/rpassword/latest/rpassword/
+///
+/// CHANGES TO THE ORIGINAL CODE:
+/// - {prompt,read}_password_from_bufread deleted (we don't need them)
+/// - rtool_box::print_tty was inlined
+/// - SafeString was removed and replaced with more general Secure<> type (and moved to cutils);
+///   also, the original code actually allowed the string to quickly escape the security net
+/// - instead of String, password are read as [u8]
+/// - this also removes the need for 'fix_line_issues', since we know we read a \n
+use std::io::{self, BufRead, Write};
+use std::mem;
+use std::os::unix::io::AsRawFd;
+
+use libc::{c_int, tcsetattr, termios, ECHO, ECHONL, TCSANOW};
+
+use sudo_cutils::Secure;
+
+struct HiddenInput {
+    fd: i32,
+    term_orig: termios,
+}
+
+impl HiddenInput {
+    fn new(fd: i32) -> io::Result<HiddenInput> {
```

Comment:
Maybe we can use `impl AsRawFd`  for our function arguments and the RawFd typedef in the struct to make it a little more clear what we are expecting here.

Reply:
Probably easier to just use the std::os::fd types.